lib/src/analysis_options.dart

```diff
@@ -115,6 +115,35 @@ String updatePassthroughOptions({
     }
   }
 
+  final origLinter = origMap['linter'];
+  if (origLinter is Map) {
+    final origRules = origLinter['rules'];
+    if (origRules is Map) {
```

Comment:
You could probably do something with patterns here:

```
if (origMap case {'linter': {'rules': Map origRules}}) {
```

I find it slightly clearer...